gpflow/optimizers/scipy.py

```diff
@@ -39,6 +39,7 @@ def minimize(
         step_callback: Optional[StepCallback] = None,
         compile: bool = True,
         allow_unused_variables: bool = False,
+        tffun_args: Dict[str, Any] = {},
```

Comment:
1. More general to allow any mapping, not just dicts.
2. Generally best to avoid mutable default arguments, since if you mutate them in (or outside) the function they end up being changed for all subsequent calls. For dicts it's conventional to use None (since there isn't a built in immutable dict structure):

```suggestion
        tffun_args: Optional[Mapping[str, Any]] = None,
    ....
        if ttfun_args is None:
            ttfun_args = {}
```
4.

Reply:
I reasoned that `tffun_args` wasn't ever going to be mutated; but I agree it is better to use None. I have made both those suggested improvements.